packages/curve-ui-kit/src/features/user-profile/store.ts

```diff
@@ -0,0 +1,106 @@
+import type { Locale } from '@ui-kit/lib/i18n'
+import type { ThemeKey } from '@ui-kit/themes/basic-theme'
+
+type State = {
+  theme: ThemeKey
+  locale: Locale['value']
+  /** Key is either 'global' or a chainIdPoolId from getChainPoolIdActiveKey. */
+  maxSlippage: { global: string } & Partial<Record<string, string>>
+  isAdvancedMode: boolean
+}
+
+type Action = {
+  reset: () => void
+  setTheme: (theme: ThemeKey) => void
+  setLocale: (locale: Locale['value']) => void
+  /**
+   * Sets or removes a max slippage value for a given key.
+   * @param slippage - The slippage value as a string percentage (e.g. "0.1" for 0.1%), or null to remove
+   * @param key - Optional key to set slippage for (e.g. "router" or chainId-poolId). If omitted, sets slippage to all existing keys.
+   * @returns boolean - True if slippage was successfully set/removed, false if invalid input
+   * @example
+   * // Set router slippage to 0.1%
+   * setMaxSlippage("0.1", "router")
+   *
+   * // Remove slippage for specific pool
+   * setMaxSlippage(null, "1-0x123...")
+   */
+  setMaxSlippage: (slippage: string | null, key?: string) => boolean
+  setAdvancedMode: (isAdvanced: boolean) => void
+}
+
+type Store = State & Action
+
+const INITIAL_THEME =
+  typeof window !== 'undefined'
+    ? window.matchMedia('(prefers-color-scheme: dark)').matches
+      ? 'dark'
+      : 'light'
+    : 'light'
+
+const INITIAL_STATE: State = {
+  theme: INITIAL_THEME,
+  locale: 'en',
+  maxSlippage: { global: '0.1' },
+  isAdvancedMode: false,
+}
+
+const store: StateCreator<Store> = (set) => ({
+  ...INITIAL_STATE,
+  reset: () => set(INITIAL_STATE),
+  setTheme: (theme) => set((state) => ({ ...state, theme })),
+  setLocale: (locale) => set((state) => ({ ...state, locale })),
+
+  setMaxSlippage: (maxSlippage: string | null, key?: string) => {
+    // Check if we want to delete a slippage value first.
+    if (maxSlippage === null) {
+      if (!key) return false
+      if (key === 'global') return false
+
+      set(
+        produce((state) => {
+          delete state.maxSlippage[key]
+        }),
+      )
+
+      return true
+    }
+
+    // Instead, we want to add or modify a slippage value.
+    const slippage = Number(maxSlippage)
+    if (isNaN(slippage) || slippage <= 0) return false
+
+    // Set slippage for a key, but if none given all existing keys will be overwritten.
+    set(
+      produce((state) => {
+        if (key) {
+          state.maxSlippage[key] = maxSlippage
+        } else {
+          for (const k of Object.keys(state.maxSlippage)) {
+            state.maxSlippage[k] = maxSlippage
+          }
+        }
+      }),
+    )
+
+    return true
+  },
+
+  setAdvancedMode: (isAdvancedMode) => set((state) => ({ ...state, isAdvancedMode })),
+})
+
+const cache: PersistOptions<Store> = {
+  name: 'user-profile',
+  merge: (persistedState, currentState) => merge(currentState, persistedState),
+  version: 1,
+}
+
+const useUserProfileStore =
```

Comment:
Ah cool so apparently they use global data rather than context. IMO using "profile" here isn't the right name but I'm not super strong on it
```suggestion
const useUserPreferences =
```

Reply:
Not super strong on it either, but for the moment it's 'coupled' with the user profile setting so I'd rather keep until it's no longer suitable in the future.